libraries/chain/include/eosio/chain/thread_utils.hpp

```diff
@@ -201,6 +205,13 @@ namespace eosio { namespace chain {
          }
       }
 
+      on_except_t make_on_except_abort() {
+         return [tn=thread_name_base_from_typeinfo(typeid(this))](const fc::exception& e) {
+            elog("Unexpected exception in a ${n} thread, aborting: ${e}", ("n", tn)("e", e.to_detail_string()));
+            abort();
+         };
+      }
+
```

Comment:
I wanted to make a `make_on_except_quit_appbase()` and then replace a number of usages of that across the code but libchain does not depend on appbase so I can't do that here. oh well :disappointed: